crates/router/src/core/payment_methods/utils.rs

```diff
@@ -0,0 +1,714 @@
+        cgraph::NodeValue::Value(dir::DirValue::PaymentType(
+            euclid::enums::PaymentType::NewMandate,
+        )),
+        None,
+        None::<()>,
+    );
+    let connectors_from_config: Vec<dir::DirValue> = eligible_connectors
+        .into_iter()
+        .filter_map(|connector| {
+            match api_enums::RoutableConnectors::from_str(connector.to_string().as_str()) {
+                Ok(connector) => Some(dir::DirValue::Connector(Box::new(
+                    api_models::routing::ast::ConnectorChoice {
+                        connector,
+                        #[cfg(not(feature = "connector_choice_mca_id"))]
+                        sub_label: None,
+                    },
+                ))),
+                Err(_) => None,
+            }
+        })
+        .collect();
+
+    if connectors_from_config.is_empty() {
+        Ok(None)
+    } else {
+        let connector_in_aggregator = builder
+            .make_in_aggregator(connectors_from_config, None, None::<()>)
+            .map_err(KgraphError::GraphConstructionError)?;
+        Ok(Some(
+            builder
+                .make_all_aggregator(
+                    &[
+                        (
+                            payment_type_value_node,
+                            cgraph::Relation::Positive,
+                            cgraph::Strength::Strong,
+                        ),
+                        (
+                            connector_in_aggregator,
+                            cgraph::Relation::Positive,
+                            cgraph::Strength::Strong,
+                        ),
+                    ],
+                    None,
+                    None::<()>,
+                    None,
+                )
+                .map_err(KgraphError::GraphConstructionError)?,
+        ))
+    }
+}
+
+// fn construct_card_network_nodes(
+//     builder: &mut cgraph::ConstraintGraphBuilder<'_, dir::DirValue>,
+//     mca_card_networks: Vec<api_enums::CardNetwork>,
+// ) -> Result<Option<cgraph::NodeId>, KgraphError> {
+//     Ok(Some(
+//         builder
+//             .make_in_aggregator(
+//                 mca_card_networks
+//                     .into_iter()
+//                     .map(dir::DirValue::CardNetwork)
+//                     .collect(),
+//                 None,
+//                 None::<()>,
+//             )
+//             .map_err(KgraphError::GraphConstructionError)?,
+//     ))
+// }
+
+fn compile_accepted_countries_for_mca(
+    builder: &mut cgraph::ConstraintGraphBuilder<'_, dir::DirValue>,
+    payment_method_type: &enums::PaymentMethodType,
+    pm_countries: Option<admin::AcceptedCountries>,
+    config: &settings::ConnectorFilters,
+    connector: String,
+) -> Result<Option<cgraph::NodeId>, KgraphError> {
+    let mut agg_nodes: Vec<(cgraph::NodeId, cgraph::Relation, cgraph::Strength)> = Vec::new();
+    if let Some(pm_obj_countries) = pm_countries {
```

Comment:
We shouldn't have an `if/else` here. You basically have
```rust
if mca_countries {
  add_mca_countries
} else if config_countries {
  add_config_countries
}
```
You're only adding one of the two, not both. It should be
```rust
if mca_countries {
  add_mca_countries
}

if config_countries {
  add_config_countries
}
```
Ensure this for currencies as well.